backend/src/main/java/de/amos/apachepulsarui/controller/NamespaceController.java

```diff
@@ -29,15 +30,32 @@ public class NamespaceController {
     private final TenantService tenantService;
 
     @GetMapping("/all")
-    public ResponseEntity<NamespacesDto> getAllNames() {
-        List<String> tenants = tenantService.getAllNames();
-        List<String> namespaces = namespaceService.getAllNames(tenants);
-        return new ResponseEntity<>(new NamespacesDto(namespaces), HttpStatus.OK);
+    public ResponseEntity<NamespacesDto> getAll(@RequestParam(required = false, defaultValue = "") List<String> tenants,
+                                                @RequestParam(required = false, defaultValue = "") List<String> namespaces) {
+        if (!namespaces.isEmpty()) {
```

Comment:
This List is null if not given in the request. So you will need a null check here.